StageMode Orbit should decelerate on release of pinch

Phipson opened the pull request:
#### a41360c4c68adf6e2426e5f530531d6563fbafaf
<pre>
StageMode Orbit should decelerate on release of pinch
<a href="https://bugs.webkit.org/show_bug.cgi?id=288738">https://bugs.webkit.org/show_bug.cgi?id=288738</a>
<a href="https://rdar.apple.com/144645278">rdar://144645278</a>

Reviewed by Wenson Hsieh.

This PR modifies the StageModeInteractionDriver by tracking the pinch velocity of the modelInteractionPanGestureRecognizer
while orbit is performed, to then apply a separate animation on the turntableContainer to decelerate the rotation
after the pinch ends. We apply a separate animation onto the turntableContainer that is triggered through a proxy. If the
velocity is reduced below a certain threshold before the animation ends, we perform an early return and terminate the interaction

* Source/WebKit/WebKitSwift/StageMode/WKStageMode.swift:
(WKStageModeInteractionDriver.currentOrbitVelocity):
(WKStageModeInteractionDriver.yawDecelerationAnimationController):
(WKStageModeInteractionDriver.yawAnimationIsPlaying):
(WKStageModeInteractionDriver.interactionDidBegin(_:)):
(WKStageModeInteractionDriver.interactionDidEnd):
(WKStageModeInteractionDriver.stageModeInteractionInProgress):
(WKStageModeInteractionDriver.subscribeToPitchChanges):
(WKStageModeInteractionDriver.subscribeToYawChanges):
- Apply a new animation to the yaw/turntableContainer for stagemode that decelerates the rotation over time

Canonical link: <a href="https://commits.webkit.org/291412@main">https://commits.webkit.org/291412@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1e770f3ab57bc825ab2b486b20a5caf888477aa9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/92764 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/12315 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/1955 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/97760 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/43275 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/12595 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/20767 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/71010 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/28437 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/95766 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/9521 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/83955 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/51339 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/9211 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/42603 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/79521 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/1558 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/99782 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/19816 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/14573 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/80015 "Found 1 new test failure: imported/w3c/web-platform-tests/service-workers/service-worker/redirected-response.https.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/20067 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/79849 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/79317 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19695 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/23846 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/1111 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/12830 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/19800 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/24976 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/19487 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/22947 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/21228 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/WebKitSwift/StageMode/WKStageMode.swift:
```diff
@@ -0,0 +1,267 @@
+import RealityKit
+import Spatial
+import Foundation
+ @_spi(RealityKit) import RealityKit
```
extra space before the @_spi

Phipson committed:
StageMode Orbit should decelerate on release of pinch
https://bugs.webkit.org/show_bug.cgi?id=288738
rdar://144645278

Reviewed by Wenson Hsieh.

This PR modifies the StageModeInteractionDriver by tracking the pinch velocity of the modelInteractionPanGestureRecognizer
while orbit is performed, to then apply a separate animation on the turntableContainer to decelerate the rotation
after the pinch ends. We apply a separate animation onto the turntableContainer that is triggered through a proxy. If the
velocity is reduced below a certain threshold before the animation ends, we perform an early return and terminate the interaction

* Source/WebKit/WebKitSwift/StageMode/WKStageMode.swift:
(WKStageModeInteractionDriver.currentOrbitVelocity):
(WKStageModeInteractionDriver.yawDecelerationAnimationController):
(WKStageModeInteractionDriver.yawAnimationIsPlaying):
(WKStageModeInteractionDriver.interactionDidBegin(_:)):
(WKStageModeInteractionDriver.interactionDidEnd):
(WKStageModeInteractionDriver.stageModeInteractionInProgress):
(WKStageModeInteractionDriver.subscribeToPitchChanges):
(WKStageModeInteractionDriver.subscribeToYawChanges):
- Apply a new animation to the yaw/turntableContainer for stagemode that decelerates the rotation over time

Canonical link: https://commits.webkit.org/291412@main
